### ISSUE TITLE
SIP GetStarted example not working

### ISSUE THREAD

**Issue by FabulousGee** [sipsorcery/sipsorcery#304]
Hey there,

I am new to this topic and wanted to dive right into it with the GetStarted example:
https://github.com/sipsorcery/sipsorcery/blob/master/examples/SIPExamples/GetStarted/Program.cs

Unfortunately, I get an error stating "SIPSorceryMedia.Windows" can't be found.
I of course installed the libraries as per documentation...

Has there been any change recently?

**Comment by sipsorcery** [comment]
You're right in the middle of me making a switch to a new approach for working with audio/video/devices on Windows.

The package you're missing has been [published](https://www.nuget.org/packages/SIPSorceryMedia.Windows/) but it may not be showing up in the nuget index for you yet.

I'm about to sort out the SIP examples now so they should be fixed in a little while, or worst case tomorrow.

**Comment by FabulousGee** [comment]
This was the one I was missing - didn't even look at the NuGet Manager, so sorry for this!

With the package it works like a charm. Thanks for the fast reply!

**Comment by FabulousGee** [comment]
@sipsorcery sorry to bother you again but again I am quite stuck with some strange errors after updating to the latest version... Are you on it?
I just set up my project from scratch again but it didn't help. Error is quite non-sense / useless. It seems like there is some discrepancy between dependencies.


btw. it is quite hard for me as a beginner to follow your examples. They all seem to use different approaches and I am quite stuck at fiddling out which parts are necessary and which are not. The UserAgentClient, UserAgentServer, UserAgentRegister and SIPCallServer are confusing me every single time.

Basically I want to have the following functionality:
- place a call to a SIP-destination
- receive a call from another SIP-client/telephone

For registration I use a FritzBox (SIP-Registrar/SIP-Server).

**Comment by sipsorcery** [comment]
Make sure you are using the latest nuget packages. As part of normal development they were briefly out of sync but that was fixed this morning:

 - SIPSorcery 4.0.77-pre
 - SIPSorceryMedia.Windows 0.0.19-pre

I've updated the examples to use the latest packages so if you checkout from 'master' it will all work.

> btw. it is quite hard for me as a beginner to follow your examples. They all seem to use different approaches and I am quite stuck at fiddling out which parts are necessary and which are not. The UserAgentClient, UserAgentServer, UserAgentRegister and SIPCallServer are confusing me every single time.

OK thanks for the feedback. The flip side is that people are regularly requesting new examples. It's a bit of a dilemma. Perhaps I'll separate the examples into basic and advanced. I will think about it.

> Basically I want to have the following functionality:
>
> place a call to a SIP-destination
> receive a call from another SIP-client/telephone

The examples you should refer to are:

 - Place call [Get Started](https://github.com/sipsorcery/sipsorcery/tree/master/examples/SIPExamples/GetStarted),
 - Receive call [Record Incoming Call](https://github.com/sipsorcery/sipsorcery/tree/master/examples/SIPExamples/RecordIncomingCall) skip the recording bit,
 - Registration [User Agent REgister](https://github.com/sipsorcery/sipsorcery/tree/master/examples/SIPExamples/UserAgentRegister).

I'd also recommend having a look at this [overview](https://sipsorcery.github.io/sipsorcery/articles/sipuseragent.html) article which I literally updated a few hours and explains the concepts in the examples.

**Comment by FabulousGee** [comment]
Pretty weird. I made sure to update to the latest packages this time - just before I reopened here. Versions didn't change either but it's working again now. Maybe I added a wrong (old) lib and didn't notice - I don't know.

Thanks anyway for your super fast and detailed response! Can not describe how awesome this is and how much I appreciate!
I will look into the examples you mentioned. Seems like I was already digging to deep...
